src/socket-message-handling.ts

```diff
@@ -76,49 +158,15 @@ const setupWSConnection = (conn) => {
   conn.on(
     "message",
     /** @param {object} message */ (message) => {
-      if (typeof message === "string") {
-        message = JSON.parse(message);
-      }
-      if (message && message.type && !closed) {
-        switch (message.type) {
-          case "subscribe":
-            /** @type {Array<string>} */ (message.topics || []).forEach(
-              (topicName) => {
-                if (typeof topicName === "string") {
```

Comment:
be careful about combining refactors and features into the same commit. Keep old ideas and new ideas separate.